src/common/services/session/session.service.js

```diff
@@ -46,95 +76,297 @@ const session = /* @ngInject */ function ($cookies, $rootScope, $http, $timeout,
+      }
+      const errors = errorObject?.data?.error
+        ? errorObject.data.error.split(',')
+            .filter((str) => str.includes(':errorSummary=>') && !str.includes('Api validation failed: login'))
+            .map((str) => str.match(/"([^"]+)"/)[1].replace(/["]/g, ''))
+        : errorObject
+
+      let checkIfAccountIsPending = false
+
+      const formattedErrors = errors.map((error) => {
+        switch (error) {
+          case 'login: An object with this field already exists in the current organization':
+            checkIfAccountIsPending = true
+            return 'OKTA_EMAIL_ALREADY_EXISTS'
+          case 'email: Does not match required pattern':
+            return 'OKTA_ERROR_WHILE_SAVING_EMAIL'
+          case 'Something went wrong. Please try again':
+            return 'OKTA_ERROR_WHILE_SAVING_DATA'
+          default:
+            return error
+        };
+      })
+      if (!checkIfAccountIsPending) {
+        return {
+          status: 'error',
+          data: formattedErrors,
+          accountPending: false
         }
-      }))
-      .map((response) => response.data)
+      } else {
+        const accountPending = await checkCreateAccountStatus(email)
+        if (accountPending?.data?.status !== 'PROVISIONED') {
+          return {
+            status: 'error',
+            data: formattedErrors,
+            accountPending: false
+          }
+        } else {
+          $cookies.put(
+            createAccountDataCookieName,
+            dataAsString,
+            {
+              path: '/',
+              domain: isTest ? '' : cookieDomain,
+              expires: moment().add(2, 'hours').toISOString()
+            }
+          )
+          return {
+            status: 'error',
+            data: formattedErrors,
+            accountPending: true
+          }
+        }
+      }
+    } catch {
+      return {
+        status: 'error',
+        data: ['SOMETHING_WENT_WRONG']
+      }
+    }
   }
 
-  function forgotPassword (email, passwordResetUrl) {
-    // https://github.com/CruGlobal/cortex_gateway/wiki/Send-Forgot-Password-Email
-    return Observable
-      .from($http({
-        method: 'POST',
-        url: casApiUrl('/send_forgot_password_email'),
-        withCredentials: true,
-        data: {
-          email: email,
-          passwordResetUrl: passwordResetUrl
+  async function checkCreateAccountStatus (email) {
+    const isAuthenticated = await authClient.isAuthenticated()
+    if (currentRole() !== Roles.public || isAuthenticated) {
+      return 'Already logged in.'
+    }
+    try {
+      const createAccountStatus = await $http({
+        method: 'GET',
+        url: `${oktaApiUrl('status')}?email=${encodeURIComponent(email)}`,
+        withCredentials: true
+      })
+      return {
+        status: 'success',
+        data: createAccountStatus.data
+      }
+    } catch (err) {
+      try {
+        if (err.status === 401) {
+          throw new Error()
```

Comment:
Should we return an error instead of throwing?

```suggestion
          return { status: 'error', data: ['FORBIDDEN'] }
```